packages/driver/src/cypress/runner.ts

```diff
@@ -492,7 +492,57 @@ const hasOnly = (suite) => {
   )
 }
 
-const normalizeAll = (suite, initialTests = {}, setTestsById, setTests, getRunnableId, getHookId, getOnlyTestId, getOnlySuiteId, createEmptyOnlyTest) => {
+const pruneSuite = (emptySuite) => {
+  emptySuite.cleanReferences()
+
+  if (emptySuite.parent) {
+    const parentSuites = emptySuite.parent.suites.filter((suite) => suite !== emptySuite)
+
+    emptySuite.parent.suites = parentSuites
+    emptySuite.parent._onlySuites = emptySuite.parent._onlySuites.filter((suite) => parentSuites.includes(suite))
+    emptySuite.parent = null
+  }
+}
+
+const pruneEmptySuites = (rootSuite, incrementFoundTestsBy) => {
```

Comment:
Can you add comments on this to describe what this is doing and the expected out come. The runner code is always a lot to process

Reply:
Done in [5e431a0](https://github.com/cypress-io/cypress/pull/25265/commits/5e431a05532e2116e048ad3468ede9aa41ff6eda)!